ksqldb-engine/src/main/java/io/confluent/ksql/services/SandboxedSchemaRegistryClient.java

```diff
@@ -36,21 +43,213 @@
 
   static SchemaRegistryClient createProxy(final SchemaRegistryClient delegate) {
     Objects.requireNonNull(delegate, "delegate");
-
-    return LimitedProxyBuilder.forClass(SchemaRegistryClient.class)
-        .swallow("register", anyParams(), 123)
-        .swallow("getId", anyParams(), 123)
-        .forward("getAllSubjects", methodParams(), delegate)
-        .forward("getSchemaById", methodParams(int.class), delegate)
-        .forward("getLatestSchemaMetadata", methodParams(String.class), delegate)
-        .forward("getSchemaBySubjectAndId", methodParams(String.class, int.class), delegate)
-        .forward("testCompatibility",
-            methodParams(String.class, ParsedSchema.class), delegate)
-        .swallow("deleteSubject", methodParams(String.class), Collections.emptyList())
-        .forward("getVersion", methodParams(String.class, ParsedSchema.class), delegate)
-        .build();
+    return new SandboxSchemaRegistryCache(delegate);
   }
 
   private SandboxedSchemaRegistryClient() {
   }
+
+  static final class SandboxSchemaRegistryCache implements SchemaRegistryClient {
+    private final MockSchemaRegistryClient mockedClient = new MockSchemaRegistryClient();
```

Comment:
Can we add comments explaining the difference between `delegate` and `mockedClient` here, i.e., what each is used for/represents? Without this context, this code is very difficult to understand.

We might also want to rename `mockedClient` to `sandboxCacheClient` or similar, in order to clarify within the code itself as well.